server/index.js

```diff
@@ -69,6 +69,118 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!cartId) {
+    return res.status(200).json([]);
+  }
+  const sql = `
+    select "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      FROM "cartItems" AS "c"
+      JOIN "products" AS "p" USING ("productId")
+     WHERE "c"."cartId" = $1
+  `;
+  const params = [cartId];
+  db.query(sql, params)
+    .then(result => {
+      const cart = result.rows[0];
+      if (!cart) {
+        next(new ClientError(`Cannot find cart with "cartId" ${cartId}`, 404));
+      } else {
+        res.status(200).json(cart);
+      }

```

Comment:
The `cart` is essentially an array of items. So this endpoint should respond with all rows retrieved by the query (even if there are none).